### PR TITLE
tpp: deprecate

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
Given that this doesn't build on any version of macOS we support, I suspect the only reason this even builds on Linux is because we are using such an old compiler.  @Homebrew/core is it time to deprecate this?  `ncurses-ruby` has not been updated since 2011, and it seems `tpp` hasn't been updated since 2007, so this does not appear to be actively maintained.

**Comment by cho-m** [comment]
For build failure, Fedora's patch may help: https://src.fedoraproject.org/rpms/ruby-ncurses/blob/rawhide/f/0001-STR2CSTR-deprecated.patch

Probably worth deprecating too. I expect there are problems with Ruby 1.9+ and security issues.

Looking at temporary GitHub repo that author created after 1.3.1 and then archived (https://github.com/akrennmair/tpp), I see:
- Some commits for fixing Ruby 1.9 support. Debian, Fedora, and AUR also carry patches for Ruby 1.9.
- CVE-2013-2208 (https://github.com/akrennmair/tpp/pull/2) and another security-related commit (https://github.com/akrennmair/tpp/commit/4b3c075dd21658406d31074509faa658e84cccfa).

**Comment by MikeMcQuaid** [comment]
👍🏻 to deprecate.

File: Formula/tpp.rb [hunk 1]
```diff
@@ -18,6 +18,8 @@ class Tpp < Formula
     sha256 cellar: :any_skip_relocation, el_capitan:  "25e92e9f229433131cc82cf48a3cec90d19a28a08a56fadcc095b1ecf4df2304"
   end
 
+  deprecate! date: "2022-04-14", because: :repo_archived
```
Comment (Bo98) [inline Formula/tpp.rb:21]: We don't list a repo anywhere in this formula, so probably better to go for `:unmaintained`.

(Commit message also isn't quite right now).

Comment (danielnachun) [reply Formula/tpp.rb]: This should be ready to go now.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**